cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -1473,9 +1473,11 @@ aggregate_reader_metadata::select_columns(
 
     // Find which of the selected paths are valid and get their schema index
     std::vector<path_info> valid_selected_paths;
-    // vector reference pushback (*use_names). If filter names passed.
+    auto const empty_names = std::vector<std::string>{};
+    // vector reference pushback for select column and/or filter names passed. Use empty names
+    // reference if the no filter column names provided.
     std::vector<std::reference_wrapper<std::vector<std::string> const>> const column_names{
-      *use_names, *filter_columns_names};
+      *use_names, filter_columns_names.has_value() ? *filter_columns_names : empty_names};
```

Comment:
Do not dereference the optional `filter_column_names` reference if empty. Allows directly calling it with (std::nullopt) rvalue.